milli/tests/search/mod.rs

```diff
@@ -53,9 +56,21 @@ pub fn setup_search_index_with_criteria(criteria: &[Criterion]) -> Index {
     let mut builder = UpdateBuilder::new(0);
     builder.max_memory(10 * 1024 * 1024); // 10MiB
     let mut builder = builder.index_documents(&mut wtxn, &index);
-    builder.update_format(UpdateFormat::JsonStream);
     builder.enable_autogenerate_docids();
-    builder.execute(CONTENT.as_bytes(), |_, _| ()).unwrap();
+    let mut cursor = Cursor::new(Vec::new());
+    let mut documents_builder =
+        DocumentsBuilder::new(&mut cursor, bimap::BiHashMap::new()).unwrap();
+    let reader = Cursor::new(CONTENT.as_bytes());
+    for doc in serde_json::Deserializer::from_reader(reader).into_iter::<serde_json::Value>() {
+        documents_builder.add_documents(doc.unwrap()).unwrap();
+    }
+    documents_builder.finish().unwrap();
+
+    cursor.set_position(0);
```

Comment:
I am not sure you need to wrap the vec into a cursor [as `&mut Vec<u8>` implements write](https://doc.rust-lang.org/std/vec/struct.Vec.html#impl-Write). And [you can read it by using a `&[u8]`](https://doc.rust-lang.org/std/primitive.slice.html#impl-Read), but that's maybe easier to just wrap it into a `cursor`.

Reply:
It doesn't implement Seek